controllers/bpfman-agent/common.go

```diff
@@ -217,7 +217,7 @@ func (r *ReconcilerCommon) reconcileBpfProgram(ctx context.Context,
 		id, err := bpfmanagentinternal.GetID(bpfProgram)
 		if err != nil {
 			r.Logger.Error(err, "Failed to get bpf program ID")
-			return bpfmaniov1alpha1.BpfProgCondNotLoaded, nil
+			return bpfmaniov1alpha1.BpfProgCondNotLoaded, err
```

Comment:
If the ID didn't get set by now, how is it going to get set?  I don't think it is.  Did the users scenario hit this case?  However, we think the program is loaded becasue it's in the loadedBpfPrograms list, and the element in that list is the gobpfman.ListResponse_ListResult returned by bpfman, and that has all the info on the program, including the ID.  So, maybe we should try to pull it out of there.